### PR TITLE
VW: raise minimum steering speed to fix a fault

### LINKED ISSUE #30495: Low speed steering issue is back with 22' VW Arteon (US model).
### Describe the bug

I've been using Openpilot for about a year now, and originally run into a problem when I first started using it. I submitted a bug report, and user/dev jyoung8607 fix it in a custom branch for me. That branch was eventually merged into the following release. After the last release, the issue has come back. When the system tries to steer at very low speeds I get "Steering temporarily unavailable" shows up and steering is disabled until I restart the car. See on both these drives:

c34ef43981b6d888|2023-11-19--12-31-55--0
c34ef43981b6d888|2023-11-19--11-52-54--0

### Which car does this affect?

VW Arteon 2022 (US model)

### Provide a route where the issue occurs

c34ef43981b6d888|2023-11-19--12-31-55--0

### openpilot version

openpilot 0.9.4

### Additional info

See bug description

### PR DESCRIPTION
**Pull request by jyoung8607**
**Description**

Resolves #30495.

**Fix**

Slightly increased Volkswagen default `minSteerSpeed`, from the [effective global default of 0.3 m/s](https://github.com/commaai/openpilot/blob/eada51ec8ed570a4efe70909760460c976cd4f04/selfdrive/controls/lib/latcontrol.py#L6) up to 0.4 m/s. Since we already considered 0.67 mph as "steer to zero", tweaked the doc generator to show the same for VW's 0.89 mph.

We could definitely do this only for the affected EPS rack-families, but we won't get EPS firmware for installs without comma power, and getting back 0.1 m/s doesn't seem worth adding any complexity. So, this will just apply to all VW.

**Verification**

Test-driven by affected users.

**Route**

Route: `f39cf149898833ff/2024-02-15--17-27-50/0`

### CODE REVIEW

File: selfdrive/car/docs_definitions.py [hunk 1]
```diff
@@ -269,6 +269,8 @@ def init(self, CP: car.CarParams, all_footnotes: Dict[Enum, int]):
       assert CP.minSteerSpeed == 0, f"{CP.carFingerprint}: Minimum steer speed set in both CarInfo and CarParams"
     else:
       self.min_steer_speed = CP.minSteerSpeed
+      if self.min_steer_speed < 1.0 * CV.MPH_TO_MS:
+        self.min_steer_speed = 0
```
Comment (jyoung8607) [inline selfdrive/car/docs_definitions.py:273]: @sshane You had mentioned doing the rounding in `__post_init__` but this location feels logical to me. Let me know if you think it's a problem.

**Comment by dukeajuke** [comment]
jyoung8607 - I installed your branch with the fix in it last night and tested it, and it worked like a champ. Instead of causing the fault error when it pulls away from a stop and tries to turn, it displays "Steering unavailable under 1 mile per hour" or something similar and motors on.

**Comment by jyoung8607** [comment]
Glad to hear it's functionally effective, but I intended for openpilot to stay silent (not alert about low speed) as well. There's one more alert I need to tweak. I'll try to push an update for that in the next couple days.

**Comment by jyoung8607** [comment]
@dukeajuke Check for updates and try it again, it shouldn't display the low speed alert anymore.

**Comment by Saber422** [comment]
@jyoung8607 I saw this PR and test this branch on my 2023 Kodiaq RS (Europe import to Taiwan). Kodiaq RS's EPS seems more critical than other VW cars. This it the route shared with you. thank you. 
0bbe367c98fa1538/2024-02-19--06-44-48/0

**Comment by jyoung8607** [comment]
> @jyoung8607 I saw this PR and test this branch on my 2023 Kodiaq RS (Europe import to Taiwan). Kodiaq RS's EPS seems more critical than other VW cars. This is the route shared with you. thank you. 0bbe367c98fa1538/2024-02-19--06-44-48/0

This looks like a different set of conditions entirely. You were moving at nearly 6 km/h with a high steering angle, looks like the EPS refused to steer any further than the 130 degrees you manually steered it to and eventually faulted.

I'm sure we can deal with it somehow, but I'll need to see more examples to fully understand the triggering conditions. Can you open a separate issue, add this route to it, and then add some more examples as you're able to reproduce it?

**Comment by jyoung8607** [comment]
@dukeajuke @Gyrohammer Have either of you tested the current version of this PR? Need your feedback to move forward.

**Comment by Gyrohammer** [comment]
> @dukeajuke @Gyrohammer Have either of you tested the current version of this PR? Need your feedback to move forward.

Been testing for the past few days, everything seems to be working correctly.

**Comment by jyoung8607** [comment]
Ready for review then!

The process_replay difference is expected, due to the updated minSteerSpeed parameter.

```
***** results for segment regen8BDFE7307A0|2023-10-30--23-19-36--0 *****
    controlsd
        ref: https://commadataci.blob.core.windows.net/openpilotci/regen8BDFE7307A0|2023-10-30--23-19-36--0_controlsd_7d25b1f7d0bd3b[506](https://github.com/commaai/openpilot/actions/runs/7922952955/job/21631686752?pr=31450#step:6:507)fa4e72ff893728894eb1a45.bz2
        new: /tmp/openpilot/openpilot/selfdrive/test/process_replay/fakedata/regen8BDFE7307A0|2023-10-30--23-19-36--0_controlsd_2d6a36ff6533c71bebd31bacc13c9f89f7286bfe.bz2

        carParams.minSteerSpeed: 2
    radard
    plannerd
    calibrationd
    dmonitoringd
    locationd
    paramsd
    ubloxd
    torqued
```

**Comment by dukeajuke** [comment]
I did and responded with results. It fixed the problem but displays a message "Steering unavailable under 5 mph" when it tries to steer from a stop. Jason mentioned he may try to suppress the message.



Sent from my Verizon, Samsung Galaxy smartphone



-------- Original message --------
From: Gyrohammer ***@***.***>
Date: 2/21/24 3:34 PM (GMT-05:00)
To: commaai/openpilot ***@***.***>
Cc: David Whapham ***@***.***>, Mention ***@***.***>
Subject: Re: [commaai/openpilot] VW: Steer to zero, for large values of zero (PR #31450)


@dukeajuke<https://github.com/dukeajuke> @Gyrohammer<https://github.com/Gyrohammer> Have either of you tested the current version of this PR? Need your feedback to move forward.

Been testing for the past few days, everything seems to be working correctly.

**Comment by jyoung8607** [comment]
> I did and responded with results. It fixed the problem but displays a message "Steering unavailable under 5 mph" when it tries to steer from a stop. Jason mentioned he may try to suppress the message.

We received that, and I pushed an update [later that same day](https://github.com/commaai/openpilot/pull/31450#issuecomment-1947447339) that should suppress it. Make sure you have the latest.

**Comment by adeebshihadeh** [comment]
Good to review or still a WIP?

**Comment by jyoung8607** [comment]
Functional and good to review. @sshane may have opinions on how the docs code was tweaked.

**Comment by sshane** [comment]
@jyoung8607 any higher and I think we'll have to show 1 mph in the docs :wink:

**Pull request merged**